paimon-core/src/main/java/org/apache/paimon/table/system/SchemasTable.java

```diff
@@ -223,6 +226,22 @@ public InnerTableRead withFilter(Predicate predicate) {
                         handleLeafPredicate(leaf, leafName);
                     }
                 }
+
+                // optimize for IN filter
+                if ((compoundPredicate.function()) instanceof Or) {
```

Comment:
We should extract a class to reduce duplicate codes, this can be done in a new PR.